### ISSUE TITLE
Invalid RUN language: fsharp

### ISSUE THREAD

**Issue by haf** [Shippable/support#2581]
### Is your request:

- [x] A feature request: The feature you want currently does not exist. If implemented, will provide additional value to you.
- [ ] A technical issue: You are unable to use Shippable's service per your expectation & need help to resolve the issue.
- [ ] A question: A non-technical question such as billing.

### Description of your issue:
NOTE: If you are experiencing a build failure, please include:

- Link to the build failure: https://app.shippable.com/runs/57234be0a2db4f0e00635ebf
- Scripts called from the yml (optional)

```
---
language: fsharp

fsharp: 4.0.0.4

build:
  pre_ci_boot: fsharp/fsharp
  image_tag: staging
  pull: true
  options: '-e PRODUCTION=true'

  pre_ci:
    - echo "This is the master branch"

  ci:
    - bundle
    - bundle exec rake

  post_ci:
    - ls -lah
    - pwd

  cache: true

  cache_dir_list:
    - $SHIPPABLE_BUILD_DIR/packages
```

How do I build with F# with albacore? https://github.com/Albacore/albacore/wiki

Basically: how to get started?

**Comment by kanthchandra** [comment]
Hi @haf, we dont officially support fsharp, but if your image has all the dependencies of fsharp, you can use `language: none` tag. 

If you use the language: none tag, we dont start any default services or we dont check for specific languages supported by us or setup any environment that your build needs. You will have to set it up everything yourself in the ci section. 

Do give it a try and let us know if you need any help.

**Comment by haf** [comment]
What do you recommend placing packages in? `prm`?

**Comment by kanthchandra** [comment]
Does your build image have node and npm installed? If not I would recommend installing all the dependencies in your build image like bundler, rake, npm, or anything that your build needs. It looks like albacore package is not able to find npm in the build container. 

I'm not sure what you mean by build environment freeze. If you mean caching, we do support folder level caching and you can find the docs here http://docs.shippable.com/ci_configure/#caching .

You can find docs related to configuring your build here http://docs.shippable.com/ci_configure

**Comment by haf** [comment]
Btw can anyone read the logs I posted a link to?

**Comment by haf** [comment]
Next error:

```
SHIPPABLE_IS_LEGACY_IMAGE=false fsharp/fsharp: 2s  
38b133533dc3ca1e20dcd41a728c1c208a805b781b435f24a80ee286308b9437
Error response from daemon: Cannot start container 38b133533dc3ca1e20dcd41a728c1c208a805b781b435f24a80ee286308b9437: lstat /dev/.udev.log: no such file or directory
```

Happened 16th run.

**Comment by kanthchandra** [comment]
Hi @haf , all your latest build seems to be in the green and working as expected (congrats on that! ). 

@manishas I'll keep this open as a feature request for supporting fsharp language and I'm unassigning myself.

**Comment by manishas** [comment]
I am closing this issue since the workaround for this is to use the `language: none` tag in the yml.

We will reopen if we get a lot of +1s for fsharp integration.
